### ISSUE TITLE
Self-sizing behavior of `NukeUI.LazyImage` compared to `SwiftUI.AsyncImage`

### ISSUE THREAD

**Issue by vrutberg** [kean/Nuke#610]
Hi,

First of all I would like to thank you for making this library. We are strongly considering adopting it at my work and it seems very well made and nicely documented. ⭐ 

I have a question regarding the self-sizing behavior of `NukeUI.LazyImage` compared to `SwiftUI.AsyncImage`.

Take a look at the following examples:

<details>
<summary>SwiftUI.AsyncImage</summary>

```swift
struct ContentView: View {
    private let imageUrl: URL = URL(string: "https://picsum.photos/1000")!

    var body: some View {
        VStack {
            AsyncImage(
                url: imageUrl,
                content: {
                    $0.resizable().aspectRatio(contentMode: .fit)
                },
                placeholder: { ProgressView() }
            )

            Text("Hello, world!")
        }
        .padding()
    }
}
```

<img width="300" src="https://user-images.githubusercontent.com/49580/194527339-74127998-4f13-468e-9067-468e96555220.png">
</details>

<details>
<summary>NukeUI.LazyImage</summary>

``` swift
struct ContentView: View {
    private let imageUrl: URL = URL(string: "https://picsum.photos/1000")!

    var body: some View {
        VStack {
            LazyImage(
                url: imageUrl,
                resizingMode: .aspectFit
            )

            Text("Hello, world!")
        }
        .padding()
    }
}
```

<img width="300" src="https://user-images.githubusercontent.com/49580/194526962-e79e8484-fb1d-4c4e-be00-6a4c2ac4795d.png">
</details>

In the 2nd example, when `NukeUI.LazyImage` is used, it seems to consume all available vertical space and pushes the `Hello, world!` text to the bottom of the screen. What I want is the image view to size itself according to the size of the image. In this case I know the image is 1000x1000, so I could set a fixed frame, but let's say I didn't know the size of the image. Is there a way to achieve the self-sizing behavior of `SwiftUI.AsyncImage` when using `NukeUI.LazyImage`?

If this is mentioned in the documentation and I missed it, I apologize in advance (I promise, I have looked! 😄)

**Comment by kean** [comment]
`LazyImage` wasn't designed to match `AsyncImage`. It works the same way as a `UIImageView` wrapped in `UIViewRepresentable` (because that's how it's implemented). The aspect modes get applied to the underlying `UIImageView`.

> self-sizing behavior 

It was a long time since I built, but as far as I remember, it wasn't possible because there was no way to communicate intrinsic size changes from `UIView` to `SwiftUI`. It might've changed now, and there may be workarounds. I think it's worth revisiting this and documenting it more clearly.

**Comment by kean** [comment]
Addressed in [Nuke 12.0 (Beta 1)](https://github.com/kean/Nuke/releases/tag/12.0.0-beta.1). It should now match exactly.

**Comment by kean** [comment]
Please test it in [Nuke 12 (RC1)](https://github.com/kean/Nuke/releases/tag/12.0.0-rc.1).
